sql/session.go

```diff
@@ -32,15 +32,20 @@ type Session interface {
 	GetAll() map[string]TypedValue
 	// ID returns the unique ID of the connection.
 	ID() uint32
+	// Warn stores the warning in the session.
+	Warn(warn *Warning)
```

Comment:
should we separate this into another session interface to do not force people to implement warnings?

Reply:
`type LogSession` +  `func WithWarnings`?

Reply:
If warnings is supported, it should be implemented in the session imho